recipe/meta.yaml

```diff
@@ -120,7 +146,7 @@ outputs:
         - ndarray =1
         - networkx
         - numexpr =2
-        - numpy =1.20
+        - {{ pin_compatible('numpy') }}
```

Comment:
Wait, there's another `numpy` above in the same `run` section.

Reply:
yes, you need both in general

If you have a pin like 

```
numpy =1.20
```

then it van install `1.20.1`, `1.20.2` etc. 

However, code compiled with say `1.20.2` should not be used at runtime with anything less than `1.20.2`. The current way these are pinned does not enforce that.

By having both we get the desired effect:

```
numpy {{ numpy }}
{{ pin_compatible('numpy') }}
```

for `numpy = 1.20.3` resolves to the maximum version consistent with `1.20` so we get

```
numpy 1.20.*
numpy >=1.20.3,<1.21.a0
```

etc.

Reply:
numpy is special because it doesn't have run exports so we have to list the pin-compatible by hand. Other packages like mpich have run exports so we only have to list it directly and conda build takes care of the rest.